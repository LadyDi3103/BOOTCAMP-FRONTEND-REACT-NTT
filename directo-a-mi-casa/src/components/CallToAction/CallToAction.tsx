import React from 'react';

/**
 * Componente CallToAction
 * Muestra un mensaje destacado para incentivar pedidos rápidos con entrega desde 2 horas.
 */
const CallToAction: React.FC = () => {
    return (
        <section className="cta">
            <h1>
                HAZ TU PEDIDO Y RECÍBELO DESDE <span className="hours">2 HORAS</span>
            </h1>
        </section>
    );
};

export default CallToAction;
