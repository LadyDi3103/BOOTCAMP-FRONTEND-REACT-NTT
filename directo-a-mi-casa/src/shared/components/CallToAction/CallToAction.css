.cta {
    background-color: var(--color-warn);
    text-align: center;
    padding: 20px 0;
    display: flex;
    justify-content: center;
    align-items: center;
    height: 55px;
    z-index: 0;
    position: relative;
    margin-top: 5rem;
}

.cta h1 {
    color: var(--color-text-light);
    font-size: 18px;
    letter-spacing: 0.1rem;
}

.cta span {
    background-color: var(--color-accent);
    height: 38px !important;
    border-radius: 15px;
    padding: 0 10px;
}