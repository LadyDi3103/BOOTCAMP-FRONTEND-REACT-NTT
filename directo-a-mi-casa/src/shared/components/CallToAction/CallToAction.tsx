import React from 'react';
import "./CallToAction.css"; 

const CallToAction: React.FC = () => {
    return (
        <section className="cta" aria-labelledby="cta-title">
            <h1 id="cta-title">
                HAZ TU PEDIDO Y REC√çBELO DESDE <span className="hours">2 HORAS</span>
            </h1>
        </section>
    );
};

export default CallToAction;
