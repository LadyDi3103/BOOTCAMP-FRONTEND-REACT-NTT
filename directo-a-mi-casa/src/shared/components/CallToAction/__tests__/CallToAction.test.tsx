import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import CallToAction from '../CallToAction';

describe('CallToAction Component', () => {
  it('should render the heading text correctly', () => {
    render(<CallToAction />);

    const headingElement = screen.getByText("HAZ TU PEDIDO Y RECÍBELO DESDE");
    expect(headingElement).toBeInTheDocument();
  });

  it('should contain the highlighted text "2 HORAS"', () => {
    render(<CallToAction />);


    const hoursElement = screen.getByText("2 HORAS");
    expect(hoursElement).toBeInTheDocument();
    expect(hoursElement).toHaveClass('hours');
  });

  test("Should match snapshot", () => {
    const { container } = render(<CallToAction />);
    expect(container).toMatchSnapshot();
});

test("should render a <section> element with class 'cta'", () => {
  render(<CallToAction />);
  
  const sectionElement = screen.getByRole("region");
  expect(sectionElement).toBeInTheDocument();
  expect(sectionElement).toHaveClass("cta");
});


});
