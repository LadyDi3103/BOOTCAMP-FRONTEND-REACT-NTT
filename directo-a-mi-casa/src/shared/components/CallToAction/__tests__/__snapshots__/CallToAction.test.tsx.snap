// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CallToAction Component Should match snapshot 1`] = `
<div>
  <section
    aria-labelledby="cta-title"
    class="cta"
  >
    <h1
      id="cta-title"
    >
      HAZ TU PEDIDO Y REC√çBELO DESDE 
      <span
        class="hours"
      >
        2 HORAS
      </span>
    </h1>
  </section>
</div>
`;
